src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -215,3 +216,63 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
   }
   return diff;
 }
+
+#ifdef ASSERT
+void VMATree::print_on(outputStream* out) {
+  visit_in_order([&](TreapNode* current) {
+    out->print(SIZE_FORMAT " (%s) - %s - ", current->key(), NMTUtil::tag_to_name(out_state(current).mem_tag()),
+               statetype_to_string(out_state(current).type()));
+  });
+  out->cr();
+}
+#endif
+
+VMATree::SummaryDiff VMATree::set_tag(position from, size size, MemTag tag) {
+  VMATreap::Range range = _tree.find_enclosing_range(from);
+  assert(range.start != nullptr && range.end != nullptr,
+         "Setting a memory tag must be done within existing range");
+  if (range.start == nullptr || range.end == nullptr) {
+    log_debug(nmt)("set_tag called on non-existent range: " PTR_FORMAT " - " PTR_FORMAT,
+                   from, from+size);
+    return SummaryDiff();
+  }
+
+  StateType type = out_state(range.start).type();
+  RegionData new_data = RegionData(out_state(range.start).stack(), tag);
+
+  position end = MIN2(from + size, range.end->key());
```

Comment:
I get that `key` is the address which is the key for the treap, but using "key" at places where we use the address is super confusing for the casual code reader. Can we give TreapNode an alias, for "key"? E.g. "address" or "position"?